src/main/scala/higherkindness/compendium/core/ProtocolUtils.scala

```diff
@@ -16,28 +16,96 @@
 
 package higherkindness.compendium.core
 
-import cats.effect.Sync
+import java.io.{File, PrintWriter}
+
+import cats.effect.{Resource, Sync}
 import cats.implicits._
-import higherkindness.compendium.models.Protocol
+import higherkindness.compendium.models.{IdlName, Protocol}
 import higherkindness.compendium.models.transformer.types.SchemaParseException
+import higherkindness.droste.data.Mu
+import higherkindness.skeuomorph.openapi.{schema, JsonSchemaF}
+import higherkindness.skeuomorph.openapi.ParseOpenApi.{JsonSource, parseJsonOpenApi, _}
+
+import higherkindness.skeuomorph.protobuf.{ProtobufF, Protocol => ProtobufProtocol}
+import higherkindness.skeuomorph.protobuf.ParseProto._
 import org.apache.avro.Schema
 
 trait ProtocolUtils[F[_]] {
-  def validateProtocol(protocol: Protocol): F[Protocol]
+  def validateProtocol(protocol: Protocol, schema: IdlName): F[Protocol]
 }
 
 object ProtocolUtils {
 
-  private def parser: Schema.Parser = new Schema.Parser()
+  private def parserAvro: Schema.Parser = new Schema.Parser()
+
+  private def parserProtobuf[F[_]: Sync](raw: String): F[ProtobufProtocol[Mu[ProtobufF]]] = {
+    for {
+      tmpFile <- writeTempFile(raw)
+      p <- parseProto[F, Mu[ProtobufF]].parse(
+        ProtoSource(tmpFile.getName, tmpFile.getAbsolutePath.replaceAll(tmpFile.getName, ""))
+      )
+    } yield p
+  }
+
+  private def parseOpenApi[F[_]: Sync](
+      raw: String
+  ): F[Either[Throwable, schema.OpenApi[Mu[JsonSchemaF]]]] = {
+    for {
+      tmpFile <- writeTempFile(raw)
+      _       <- parseJsonOpenApi[F, Mu[JsonSchemaF]].parse(JsonSource(tmpFile)).attempt
+      yoapi   <- parseYamlOpenApi[F, Mu[JsonSchemaF]].parse(YamlSource(tmpFile)).attempt
```

Comment:
Using `attempt` will transform `F[A]` to `F[Either[Throwable, A]]`

If I am not mistaken here, if the OpenApi schema is in JSON, it will fail :thinking: 

Also, the tests for this validation are missing :wink:

Reply:
Yes, forgot to stash that change before commit 😅  The thing is openapi schema could be provided in json or yaml, I wasn't sure to add more details in sbt-compendium config  or check both cases

Reply:
So do we want to include OpenApi support in this PR ? I am ok about doing OpenApi in another PR so this one is more atomic :clap: 

We could support both, checking if one of the passes is able to parse.

Reply:
Let me clean the openapi support and let's separate the PRs :]